### ISSUE TITLE
bad result - not converging?

### ISSUE THREAD

**Issue by maorp** [19reborn/NeuS2#8]
hi thanks for this work! im trying dtu24 and running directly the testbed.exe. im on windows with cuda 12.1 and rtx2080ti. cant reproduce the results for some reason. i installed exactly as explained. this is the result after ~18000 steps:
![image](https://github.com/19reborn/NeuS2/assets/11231999/878fb6c6-80c3-45ac-9744-0953fa816f3a)

the run cmd:
F:\dev\NeuS2\build>testbed.exe --scene F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json --config ..\configs\nerf\dtu.json
create_directory:F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json\output
create_directory: F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json\output\mesh
create_directory: F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json\output\images
founded json file: F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json
total frame: 1
tmp_json_paths size: 1
13:35:50 INFO     Loading NeRF dataset from
13:35:50 INFO       F:\dev\NeuS2\data\nerf\dtu_scan24\transform.json
13:35:50 SUCCESS  Loaded 49 images after 0s
13:35:50 INFO       cam_aabb=[min=[0.924825,-0.419931,-0.835596], max=[1.96682,1.59547,0.667613]]
13:35:50 INFO     Loading network config from: ..\configs\nerf\dtu.json
m_predict_global_movement_training_step: 0
13:35:50 INFO     GridEncoding:  Nmin=16 b=1.45242 F=2 T=2^19 L=14
m_density_network_input_width: 32delta_network m_n_pos_dims: 6
accumulated transition params num: 4
******finish init accumulation parameters*****
13:35:50 INFO     Density model: 3--[HashGrid]-->28--[FullyFusedMLP(neurons=64,layers=3)]-->1
13:35:50 INFO     Color model:   3--[Composite]-->16+16--[FullyFusedMLP(neurons=64,layers=4)]-->3
{}
m_n_weights: 2048
m_n_weights_canonical: 2048
m_n_weights_delta: 0
13:35:50 INFO       total_encoding_params=10548128 total_network_params=11268
init m_network->n_params():10559396
[[0,["density_network",3072]],[1,["rgb_network",8192]],[2,["variance_network",4]],[3,["pos_encoding",10548128]],[4,["dir_encoding",0]]]
m_n_weights: 10559396
m_n_weights_canonical: 10559396
m_n_weights_delta: 0
initialize NeuS Network!
network_params_elements: 3072
no filem_variance_network_n_params: 4
{}
m_n_weights: 12
m_n_weights_canonical: 12
m_n_weights_delta: 0
transition_n_params:4
rotation_n_params:8
13:35:51 SUCCESS  Initialized Vulkan and NGX on device #0: NVIDIA GeForce RTX 2080 Ti

also with cams debug (they are flipped for some reason...) ->
![image](https://github.com/19reborn/NeuS2/assets/11231999/4f615511-a0d8-466a-93b8-425fbb41c546)

**Comment by 19reborn** [comment]
Hi, the result looks weird. I think the cameras are correct. Could you please check the config `dtu.json`? Have you changed anything? Below is the result that was just obtained on my Windows.
![8 20](https://github.com/19reborn/NeuS2/assets/55176925/c55f22b9-f3ce-45ae-90c9-1ff643f2f691)

**Comment by maorp** [comment]
hey, didn't change anything, i downloaded dtu24 from the google drive link provided. i tried dtu.json from the configs folder and config.json in dtu24 zip (actually the same) and get the bad result.

{
	"loss": {
		"otype": "Huber"
	},
	"optimizer": {
		"otype": "Ema",
		"decay": 0.95,
		"nested": {
			"otype": "ExponentialDecay",
			"decay_start": 20000,
			"decay_interval": 10000,
			"decay_base": 0.33,
			"nested": {
				"otype": "Adam",
				"learning_rate": 1e-2,
				"beta1": 0.9,
				"beta2": 0.99,
				"epsilon": 1e-15,
				"l2_reg": 1e-6,
				"optimize_delta_params": false,
				"optimize_canonical_params": true,
				"optimize_params_components": {
					"rgb_network": true
				}
			}
		}
	},
	"encoding": {
		"otype": "HashGrid",
		"n_levels": 14,
		"n_features_per_level": 2,
		"log2_hashmap_size": 19,
		"base_resolution": 16,
		"top_resolution": 2048,
		"valid_level_scale": 0.02,
		"base_valid_level_scale": 0.2,
		"base_training_step": 200
	},
	"network": {
		"otype": "FullyFusedMLP",
		"activation": "ReLU",
		"output_activation": "None",
		"n_neurons": 64,
		"n_hidden_layers": 1
	},
	"dir_encoding": {
		"otype": "Composite",
		"nested": [
			{
				"n_dims_to_encode": 3,
				"otype": "SphericalHarmonics",
				"degree": 4
			},
			{
				"otype": "Identity",
				"n_bins": 4,
				"degree": 4
			}
		]
	},
	"rgb_network": {
		"otype": "FullyFusedMLP",
		"activation": "ReLU",
		"output_activation": "None",
		"n_neurons": 64,
		"n_hidden_layers": 2
	},
    "single_variance_network": {
        "otype": "FullyFusedMLP",
		"activation": "None",
		"output_activation": "None",
		"n_neurons": 16,
		"n_hidden_layers": 0
    },
	"distortion_map": {
		"resolution": [32, 32],
		"optimizer": {
			"otype": "ExponentialDecay",
			"decay_start": 10000,
			"decay_interval": 5000,
			"decay_end": 25000,
			"decay_base": 0.33,
			"nested": {
				"otype": "Adam",
				"learning_rate": 1e-4,
				"beta1": 0.9,
				"beta2": 0.99,
				"epsilon": 1e-8
			}
		}
	},
	"envmap": {
		"loss": {
			"otype": "RelativeL2"
		},
		"optimizer": {
			"otype": "Ema",
			"decay": 0.99,
			"nested": {
				"otype": "ExponentialDecay",
				"decay_start": 10000,
				"decay_interval": 5000,
				"decay_base": 0.33,
				"nested": {
					"otype": "Adam",
					"learning_rate": 1e-2,
					"beta1": 0.9,
					"beta2": 0.99,
					"beta3": 0.9,
					"beta_shampoo": 0.0,
					"epsilon": 1e-10,
					"identity": 0.0001,
					"cg_on_momentum": false,
					"frobenius_normalization": true,
					"l2_reg": 1e-10
				}
			}
		}
	},
	"hyperparams": {
		"linear_color": false,
		"first_frame_max_training_step": 15000,
		"mask_loss_weight": 0.0,
		"ek_loss_weight": 0.01
	} ,
	"globalmove": {
		"optimizer": {
			"otype": "Ema",
			"decay": 0.95,
			"nested": {
				"otype": "ExponentialDecay",
				"decay_start": 6000,
				"decay_interval": 25,
				"decay_base": 0.99,
				"nested": {
					"otype": "Adam",
					"learning_rate": 1e-3,
					"beta1": 0.9,
					"beta2": 0.99,
					"beta3": 0.9,
					"beta_shampoo": 0.0,
					"epsilon": 1e-10,
					"identity": 0.0001,
					"cg_on_momentum": false,
					"frobenius_normalization": true,
					"l2_reg": 1e-10
				}
			}
		}
	}
}

**Comment by drperpen** [comment]
Hi,
Same problem here on Windows 10, when I run:
./testbed.exe --scene ../data/dtu_scan24/transform.json --config ../data/dtu_scan24/config.json

![unnamed](https://github.com/19reborn/NeuS2/assets/51986483/7ea7094e-b906-44e3-90af-97f88ed64296)

**Comment by 19reborn** [comment]
Hi @maorp, @drperpen, could you please try training without GUI ( simply add `--no-gui` in the training scripts) and see whether the training loss can decrease?

**Comment by drperpen** [comment]
Thanks for your reply @19reborn 
Without gui I cannot see any improvement:

`iteration=1000 loss=0.019155`

And it doesn't get any better from there on, I let it ran until 15000 iterations.

**Comment by drperpen** [comment]
@19reborn Using `ek_loss_weight=0.0` didn't improve the loss

**Comment by 19reborn** [comment]
hi @drperpen , @maorp. I think I find where the bug is.  It's in the function `load_sdf_mlp_weight` of `nerf_network.h` (https://github.com/19reborn/NeuS2/blob/main/include/neural-graphics-primitives/nerf_network.h#L775C1-L775C1). We load the predefined SDF MLP weights for geometry initialization in this function. 
Note that we find the file in the relative path `utils/mlp_weights.txt`, so you need to directly run the code in the main folder.

**Comment by drperpen** [comment]
@19reborn Yes, that did it! Thanks a lot!
